cpp/src/groupby/sort/group_collect.cu

```diff
@@ -17,30 +17,94 @@
 #include <cudf/column/column_factories.hpp>
 #include <cudf/column/column_view.hpp>
 #include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/detail/copy_if.cuh>
 #include <cudf/detail/gather.cuh>
+#include <cudf/strings/detail/utilities.cuh>
 #include <cudf/types.hpp>
 #include <cudf/utilities/span.hpp>
 
 #include <rmm/cuda_stream_view.hpp>
 
+#include <memory>
+
 namespace cudf {
 namespace groupby {
 namespace detail {
+/**
+ * @brief Purge null entries in grouped values, and adjust group offsets.
+ */
+std::pair<std::unique_ptr<column>, std::unique_ptr<column>> purge_null_entries(
```

Comment:
Ah, I see! I was wondering why this code looks very familiar to me. :]
https://github.com/rapidsai/cudf/blob/branch-0.20/cpp/src/rolling/rolling_detail.cuh#L1084

I think it would be better to put this function in the unnamed namespace. I don't expect we're going to reuse this function.
In the future, since there is significant overlap in with the `rolling_window()` code, we might consider moving this into a common place. Perhaps a `lists/detail/utilities.hpp`.

Reply:
I vote for that---moving that function into `lists/detail/utitities.hpp|cuh`. I see much potential for it.

Reply:
So, shall we move `purge_null_entries` in this PR or we do it in future?

Reply:
I would recommend to do it in a separate PR, as it would cause changes in many other places.

Reply:
I agree with @ttnghia. Let's tackle that in a separate PR.